dot/core/helpers_test.go

```diff
@@ -19,10 +21,135 @@ import (
 	"github.com/ChainSafe/gossamer/lib/runtime/wasmer"
 	"github.com/ChainSafe/gossamer/lib/trie"
 	"github.com/ChainSafe/gossamer/lib/utils"
+	"github.com/ChainSafe/gossamer/pkg/scale"
 	"github.com/golang/mock/gomock"
 	"github.com/stretchr/testify/require"
 )
 
+func balanceKey(t *testing.T, pub []byte) (bKey []byte) {
+	t.Helper()
+
+	h0, err := common.Twox128Hash([]byte("System"))
+	require.NoError(t, err)
+	h1, err := common.Twox128Hash([]byte("Account"))
+	require.NoError(t, err)
+	h2, err := common.Blake2b128(pub)
+	require.NoError(t, err)
+	bytesToConcat := [][]byte{h0, h1, h2, pub}
+	return bytes.Join(bytesToConcat, nil)
+}
+
+// Creates test service, used now for testing txnPool but can be used elsewhere when needed
+func createTestService(t *testing.T, genesisFilePath string,
+	pubKey []byte, accountInfo types.AccountInfo, ctrl *gomock.Controller) (service *Service, encodedExtrinsic []byte) {
+	t.Helper()
+
+	gen, err := genesis.NewGenesisFromJSONRaw(genesisFilePath)
+	require.NoError(t, err)
+
+	genesisTrie, err := wasmer.NewTrieFromGenesis(*gen)
+	require.NoError(t, err)
+
+	// Extrinsic and context related stuff
+	aliceBalanceKey := balanceKey(t, pubKey)
+	encodedAccountInfo, err := scale.Marshal(accountInfo)
+	require.NoError(t, err)
+
+	genesisHeader := &types.Header{
+		StateRoot: genesisTrie.MustHash(),
+		Number:    0,
+	}
+
+	cfgKeystore := keystore.NewGlobalKeystore()
+	kp, err := sr25519.GenerateKeypair()
+	require.NoError(t, err)
+	err = cfgKeystore.Acco.Insert(kp)
+	require.NoError(t, err)
+
+	// Create state service
+	var stateSrvc *state.Service
+	testDatadirPath := t.TempDir()
+
+	// Set up block and storage state
+	telemetryMock := NewMockClient(ctrl)
+	telemetryMock.EXPECT().SendMessage(gomock.Any()).AnyTimes()
```

Comment:
In SendMessage try to make the type `telemetry.Message`

Reply:
Having trouble with this. Can I pass in an interface like that? Or do I need to instantiate it and am just being dumb somehow when I am tring

Reply:
Try something like 
```
	var k telemetry.Message
	telemetryMock.EXPECT().SendMessage(gomock.AssignableToTypeOf(k)).AnyTimes()
```